src/app/config/no-config/no-config.component.ts

```diff
@@ -184,16 +218,41 @@ export class NoConfigComponent implements OnInit {
     this.router.navigate(['/main-screen']);
   }
 
-  public increasePage(): void {
+  public async increasePage(): Promise<any> {
     this.page += 1;
+    if (this.page <= 2) {
+      if (JSON.stringify(this.opInstance) != JSON.stringify(this.octoprintNodes['other'])) {
+        this.config.octoprint.url = 'http://' + this.opInstance['url'];
+        this.manualEntry = false;
+      } else {
+        this.config.octoprint.url = 'http://localhost:80/api/';
+        this.manualEntry = true;
+      }
+    } else if (this.config.octoprint.accessToken == '' && this.page > 2) {
+      this.page = 2;
+    } else if (this.page > 2) {
+      if (this.octoprintConnection === false) {
+        await this.testOctoprintAPI().then(res => {
+          this.opApiMsg = '';
+          if (this.opInstance.name != 'other') {
+            this.config.printer.name = this.opInstance.name;
+          } else {
+            this.config.printer.name = '';
+          }
+        }, err => {
+          this.opApiMsg = 'API Error: ' + this.octoprintConnectionError;
+          this.page = 2;
+        });
+      }
+    }
```

Comment:
This will query the API 4 times (for each page after 2). I would prefer to just have the check for connection at the very end, where it is currently located. Or are there any benefits to having it here?

Reply:
it won't cause of line 234.  Once it connects once it'll stop.

Reply:
Oh I missed that if statement. Sorry.